lib/modules/eic_media/modules/eic_media_statistics/src/Controller/MediaFileDownloadController.php

```
@@ -0,0 +1,102 @@
+<?php
+
+namespace Drupal\eic_media_statistics\Controller;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\media\MediaInterface;
+use Drupal\media_entity_download\Controller\DownloadController;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Provides a controller to download media file.
+ */
+class MediaFileDownloadController extends DownloadController {
+
+  /**
+   * Eic Media statistics config settings.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $eicMediaStatisticsSettings;
+
+  /**
+   * Eic Media file statistics database storage.
+   *
+   * @var \Drupal\eic_media_statistics\FileStatisticsDatabaseStorage
+   */
+  protected $fileStatisticsStorage;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    $controller = parent::create($container);
+    $controller->eicMediaStatisticsSettings = $container->get('config.factory')->get('eic_media_statistics.settings');
+    $controller->fileStatisticsStorage = $container->get('eic_media_statistics.storage.file');
+    return $controller;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function download(MediaInterface $media) {
+    try {
+      $response = parent::download($media);
+    }
+    catch (\Exception $e) {
```

I don't really get the point of catching the exception to throw it after 🤔

---

@korhankale I am doing it in case a new module wants to override my class. But should I return nothing instead?

---

@4hardfire , @korhankale ,
What's the status on this?

---

Hi
I'd say method/class overriding doesn't have to do anything here, the exception is still "forwarded" after being catched. Generally what should be done is either we stop the execution (e.g: we don't catch the exception) or we catch it, log and return/redirect something/somewhere (especially in controller classes since they are meant to respond to a route after a user request).

In this case $response won't be defined if we let the process continue (and seems like we need it). So we don't need to catch the exception (unless we want to display something custom)

---

We can remove this try & catch, if you want to throw the error, just let it naturally done by calling the function